Objects/typeobject.c

```diff
@@ -10389,9 +10389,13 @@ supercheck(PyTypeObject *type, PyObject *obj)
         Py_XDECREF(class_attr);
     }
 
-    PyErr_SetString(PyExc_TypeError,
+    PyErr_Format(PyExc_TypeError,
                     "super(type, obj): "
-                    "obj must be an instance or subtype of type");
+                    "obj must be an instance or subtype of type. "
+                    "Got obj: '%.200s', type: '%.200s'.",
```

Comment:
Looks good for me. Should we write a test for this?

Reply:
Added tests for `supercheck` fail